src/main/java/com/github/manolo8/darkbot/gui/tree/editors/FileEditor.java

```diff
@@ -0,0 +1,83 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import com.google.gson.*;
+import eu.darkbot.api.config.ConfigSetting;
+import eu.darkbot.api.config.util.OptionEditor;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+public class FileEditor extends JButton implements OptionEditor<File> {
+    private JFileChooser fc;
+    private File file;
+    private final Consumer<JTree> stopEditing = JTree::stopEditing;
+
+    public FileEditor() {
+        setFocusable(false);
+        addActionListener(e -> stopEditing.accept((JTree) SwingUtilities.getAncestorOfClass(JTree.class, FileEditor.this)));
+        addActionListener(e -> {
+            if (fc == null) {
+                String path = file == null ? new File(System.getProperty("user.dir")).getAbsolutePath() : file.getPath();
+                fc = new JFileChooser(path) {
+                    @Override
+                    protected JDialog createDialog(Component parent) throws HeadlessException {
+                        JDialog dialog = super.createDialog(parent);
+                        dialog.setAlwaysOnTop(true);
+                        return dialog;
+                    }
+                };
+            }
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) setEditing(fc.getSelectedFile());
+        });
+
+    }
+
+    @Override
+    public void setText(String text) {
+        int sepIdx;
+        if (text != null && text.length() > 30 && (sepIdx = text.indexOf(File.separator, text.length() - 30)) != -1) {
+            super.setText(".." + text.substring(sepIdx));
+        } else {
+            super.setText(text);
+        }
+
+        setPreferredSize(new Dimension(getFontMetrics(getFont()).stringWidth(getText()) + 32, 0));
```

Comment:
This is again a straight copy, potentially make an util, like `Strings.shortFileName(String)`?

would make this get quite reduced to:
`super.setText(Strings.shortFileName(text));`